### ISSUE TITLE
On-the-fly compression in parallel

### ISSUE THREAD

**Issue by maxsatula** [maxsatula/ocp#19]
Download example:
* Split a source file on pieces (say 100M each)
* UTL_COMPRESS each piece in parallel (with specified limit of sessions, "degree of parallelism"
* once a piece is ready, transfer over network and uncompress on-the-fly locally

**Comment by maxsatula** [comment]
Alternatively (maybe as a separate issue?), TEMP usage can be limited by some command line option (specifying absolute size, % of overall TEMP available, or automatic). This will define how a source file is split by pieces

**Comment by maxsatula** [comment]
Monitor balance between CPU and network usage
* if CPU is idle, increase compression level for further pieces
* if network is idle, add more CPU threads or decrease compression level for further pieces
